x/programs/examples/imports/program/program.go

```diff
@@ -110,10 +119,13 @@ func (i *Import) callProgramFn(callContext program.Context) func(*wasmtime.Calle
 			)
 			return -1
 		}
+		otherContext := &program.Context{
+			ProgramID: ids.ID(args.ProgramID),
+		}
 
 		// create a new runtime for the program to be invoked with a zero balance.
 		rt := runtime.New(i.log, i.engine, i.imports, i.cfg)
-		err = rt.Initialize(context.Background(), callContext, programWasmBytes, engine.NoUnits)
+		err = rt.Initialize(context.Background(), otherContext, programWasmBytes, engine.NoUnits)
```

Comment:
It might be fun to write a test that calls itself recursively and does an infinite loop just to make sure that units are exhausted, what do you think ?

Reply:
💯 ! 

Out of scope for this PR though, want to create an issue?

Reply:
https://github.com/ava-labs/hypersdk/issues/899